src/components/register/steps/contact.tsx

```diff
@@ -1,18 +1,25 @@
 /** @jsxImportSource @emotion/react */
 
 import { Localized } from '@fluent/react'
+import { useEffect } from 'react'
 import { RouteComponentProps } from '@reach/router'
 import { useForm } from 'react-hook-form'
 import { TextField } from '@eurofurence/reg-component-library'
 import WithInvoiceRegisterLayout from '../layout/with-invoice'
-import { SubmitContactInfo } from '../../../state/actions/register'
+import { ChangeContactInfo, SubmitContactInfo } from '../../../state/actions/register'
 import { ContactInfo } from '../../../state/models/register'
 import { useAppDispatch } from '../../../hooks/redux'
 
 const Contact = (_: RouteComponentProps) => {
-	const { register, handleSubmit } = useForm<ContactInfo>()
+	const { register, watch, handleSubmit } = useForm<ContactInfo>()
 	const dispatch = useAppDispatch()
 
+	useEffect(() => {
+		const subscription = watch(data => dispatch(ChangeContactInfo.create(data)))
+
+		return () => subscription.unsubscribe()
```

Comment:
Since we need to return a function, can we not just return `subscription.unsubscribe`?

Reply:
Depends on whether that function has a bound `this` or not. This construct came from the react-hook-form API docs. I'll check if the unsubscribe function has a bound `this`.

Reply:
Looks like it's bound